pkg/replica/backup.go

```diff
@@ -154,24 +158,33 @@ type BackupStatus struct {
 	BackupURL     string
 	State         ProgressState
 	IsIncremental bool
+	IsOpened      bool
 }
 
-func NewBackup(backingFile *backingfile.BackingFile) *BackupStatus {
+func NewBackup(name, volumeID, snapID string, backingFile *backingfile.BackingFile) *BackupStatus {
+	backupName := name
+	if backupName == "" {
+		backupName = butil.GenerateName("backup")
+	}
 	return &BackupStatus{
+		Name:        backupName,
 		backingFile: backingFile,
-		State:       ProgressStateInProgress,
+		State:       ProgressStateStarting,
+		volumeID:    volumeID,
+		SnapshotID:  diskutil.GenerateSnapshotDiskName(snapID),
 	}
 }
 
-func (rb *BackupStatus) UpdateBackupStatus(snapID, volumeID string, progress int, url string, errString string) error {
+func (rb *BackupStatus) UpdateBackupStatus(snapID, volumeID string, state string, progress int, url string, errString string) error {
 	id := diskutil.GenerateSnapshotDiskName(snapID)
 	rb.lock.Lock()
 	defer rb.lock.Unlock()
-	if err := rb.assertOpen(id, volumeID); err != nil {
-		logrus.Errorf("Returning Error from UpdateBackupProgress")
+
+	if err := rb.assertVolumeSnapshot(id, volumeID); err != nil {
```

Comment:
nit: It is good to make imperative check in a meaningful method, but `assert` is usually used for testing purpose instead to assert something true.

Reply:
Updated. Thank you.